Task3/Project/DataControl/Services/CsvFileService.cs

```diff
@@ -1,53 +1,301 @@
+        /// <exception cref="System.ArgumentNullException">
+        /// Thrown when file configuration is null.
+        /// </exception>
         public TaxiDriver.Champion[] GetBest(int amount)
         {
-            throw new System.NotImplementedException();
+            CheckFileConfiguration();
+
+            string[] scoreLines = File.ReadAllLines(fileConfiguration.ScoreFile);
+
+            List<KeyValuePair<string, double>> allChampionsList =
+                new List<KeyValuePair<string, double>>(scoreLines.Length);
+            string[] scoreParameters;
+            for (int i = 0; i < scoreLines.Length; ++i) 
+            {
+                scoreParameters = scoreLines[i].Split(';');
+                allChampionsList.Add(new KeyValuePair<string, double>
+                    (scoreParameters[0], double.Parse(scoreParameters[scoreParameters.Length - 1])));
+            }
+            allChampionsList.Sort((first, second) => second.Value.CompareTo(first.Value));
+
+            TaxiDriver.Champion[] champions = 
+                new TaxiDriver.Champion[amount < allChampionsList.Count ? amount : allChampionsList.Count];
+            for (int i = 0; i < amount && i < allChampionsList.Count; ++i) 
+            {
+                champions[i] = new TaxiDriver.Champion(i + 1, allChampionsList[i].Key, allChampionsList[i].Value);
+            }
+            return champions;
         }
 
+        /// <summary>
+        /// Gets the random order from some files.
+        /// </summary>
+        /// <returns>
+        /// The random order.
+        /// </returns>
+        /// <exception cref="FileNotFoundException">
+        /// Thrown when some files for creating order weren`t found.
+        /// </exception>
+        /// <exception cref="System.ArgumentNullException">
+        /// Thrown when file configuration is null.
+        /// </exception>
         public TaxiDriver.Order GetRandomOrder()
         {
-            throw new System.NotImplementedException();
+            CheckFileConfiguration();
+
+            TaxiDriver.Client client = CreateClient();
+            TaxiDriver.Route route = CreateRoute();
+
+            return new TaxiDriver.Order(rand.Next(), client, route);
         }
 
-        public bool LogIn(string name, string password)
+        /// <summary>
+        /// Sets file configuration.
+        /// </summary>
+        /// <param name="configuration">Some configuration for setting up.</param>
+        /// <returns>
+        /// Current service for data access.
+        /// </returns>
+        /// <exception cref="System.ArgumentNullException">
+        /// Thrown when the configuration is null.
+        /// </exception>
+        public Interfaces.IDataAccessService SetConfiguration(Interfaces.IConfiguration configuration)
         {
-            throw new System.NotImplementedException();
+            if (configuration != null)
+            {
+                fileConfiguration = configuration as FileConfiguration;

```

Comment:
could return null
first check if configuration IS FileConfiguration
then initialize your fields